src/main/java/com/github/kokorin/jaffree/ffmpeg/FFmpeg.java

```diff
@@ -79,8 +81,46 @@ public FFmpeg setComplexFilter(FilterGraph graph) {
         return this;
     }
 
+    /**
+     * Sets the 'generic' filter value (equivalent to the "-filter" command-line parameter).
+     *
+     * @param filter a String describing the filter to apply
+     * @return this
+     */
     public FFmpeg setFilter(String filter) {
-        this.filter = filter;
+        return setFilter((String) null, filter);
+    }
+
+    /**
+     * Sets a 'stream specific' filter value (equivalent to the "-av" / "-filter:a" or "-fv" / "-filter:v" command-line parameters).
+     *
+     * @param streamType the stream type to apply this filter to (StreamType.A for audio, StreamType.V for video)
+     * @param filter     a String describing the filter to apply
+     * @return this
+     */
+    public FFmpeg setFilter(StreamType streamType, String filter) {
+        return setFilter(streamType.code(), filter);
+    }
+
+    /**
+     * Sets a 'stream specific' filter value (equivalent to the "-av" / "-filter:a" or "-fv" / "-filter:v" / "-filter" command-line parameters).
+     *
+     * @param streamSpecifier the String specifying to which stream this filter must be applied ("a" for audio, "v" "for video, or null for generic 'filter')
+     * @param filter          a String describing the filter to apply
+     * @return this
+     */
+    public FFmpeg setFilter(String streamSpecifier, String filter) {
+        if (filters.get(streamSpecifier) != null) {
+            if (streamSpecifier == null) {
```

Comment:
You mean replacing the previous filter by the new one instead of ignoring the new one ?
Makes sense, it's consistent with the current behaviour when calling `setFilter(String filter)´ several times: the last call wins.
I'll change that right now.